nebula-logger/core/main/configuration/classes/LoggerParameter.cls

```diff
@@ -340,10 +340,35 @@ public class LoggerParameter {
         return parameterValue != null ? parameterValue : defaultValue;
     }
 
-    // Private methods
+    public static List<LoggerParameter__mdt> matchOnPrefix(String developerNamePrefix) {
+        List<LoggerParameter__mdt> matchingParameters = new List<LoggerParameter__mdt>();
+        for (String parameterDeveloperName : DEVELOPER_NAME_TO_RECORD.keySet()) {
+            if (parameterDeveloperName.startsWith(developerNamePrefix) == true) {
+                matchingParameters.add(DEVELOPER_NAME_TO_RECORD.get(parameterDeveloperName));
+            }
+        }
+        return matchingParameters;
+    }
+
+    private static Map<String, LoggerParameter__mdt> loadRecords() {
+        Map<String, LoggerParameter__mdt> parameters = LoggerParameter__mdt.getAll().clone();
+        if (System.Test.isRunningTest() == true) {
+            parameters.clear();
+            for (String testContextParameterName : PARAMETERS_TO_LOAD_DURING_TESTS) {
+                if (LoggerParameter__mdt.getAll().containsKey(testContextParameterName) == true) {
```

Comment:
shouldn't this be `parameters.containsKey(testContextParameterName) == true` ?

Reply:
oh I see, you've cleared it above. seems like there could be a little optimization here as far as comparing to `parameters` prior to clearing for the test run so you don't have to keep referring to the map; on the other hand, I guess the `getAll()` map is cached, so maybe it's not a big deal

Reply:
Yeah, that's a good point - I don't think there will currently be much performance difference because right now, there's only a single `LoggerParameter__mdt` record that meets this use case, but I still like the idea of optimizing this a little more. I'll play around with a few options to see if I can optimize it a bit.